### PR TITLE
Add `Unix.sigwait` and implement `Thread` signal-related functions with `Unix`

### PR DESCRIPTION
**Pull request by xavierleroy**
The `sigwait` system call is currently exposed in the Thread interface, but it's more generally useful, so add it to `Unix`.

As explained  in the individual commits, `Thread.sigmask` is now functionally equivalent to `Unix.sigprocmask`, and `Thread.wait_signal` to the new `Unix.sigwait`, so we can implement the `Thread` functions in terms of the `Unix` functions, getting rid of some stub code in otherlibs/systhreads.

This will also help with #13348 .

### CODE REVIEW

File: otherlibs/unix/signals.c [hunk 1]
```diff
@@ -97,6 +97,19 @@ CAMLprim value caml_unix_sigsuspend(value vset)
...
 }
 
+CAMLprim value caml_unix_sigwait(value sigs)
+{
+  sigset_t set;
+  int retcode, signo;
+
+  decode_sigset(sigs, &set);
+  caml_enter_blocking_section();
+  retcode = sigwait(&set, &signo);
```
Comment (gasche) [inline otherlibs/unix/signals.c:107]: Note: in st_posix.h, the use of this function was conditioned on the HAS_SIGWAIT macro, correesponding to a AC_CHECK_FUNC([sigwait]...) check in the configure. Now it is only conditioned on the POSIX_SIGNALS macro, but the definition of POSIX_SIGNALS only checks for `sigaction` and `sigprocmask`. So now we are (newly) assuming that if those two are available, then `sigwait` also is (and, above, `sigsuspend`).

**Review by gasche — APPROVED** [review]
Looks fine to me, thanks! Approved.

File: otherlibs/unix/signals.c [hunk 2]
```diff
@@ -97,6 +97,19 @@ CAMLprim value caml_unix_sigsuspend(value vset)
...
+  caml_leave_blocking_section();
+  if (retcode == -1) caml_uerror("sigwait", Nothing);
```
Comment (ghost) [inline otherlibs/unix/signals.c]: This logic is wrong. `sigwait` either returns zero or a positive `errno`.
```suggestion
  if (retcode != 0) caml_uerror("sigwait", Nothing);
```

File: otherlibs/unix/unixLabels.mli [hunk 3]
```diff
@@ -1160,7 +1160,7 @@ type sigprocmask_command = Unix.sigprocmask_command =
   | SIG_UNBLOCK
 
 val sigprocmask : mode:sigprocmask_command -> int list -> int list
-(** [sigprocmask ~mode sigs] changes the set of blocked signals.
+(** [sigprocmask mode sigs] changes the set of blocked signals.
```
Comment (MisterDA) [inline otherlibs/unix/unixLabels.mli]: Why did you remove the tilde? It's a labelled argument.

File: otherlibs/unix/signals.c (hunk 1 above)
Comment (xavierleroy) [reply otherlibs/unix/signals.c:107]: See my helpful commit message in b62ba96d14.  This kind of per-function configure test was appropriate in the 1990's, but nowadays either you're on Windows or you have full POSIX.1-2008 (or at the very least -2001) support.

File: otherlibs/unix/signals.c (hunk 2 above)
Comment (xavierleroy) [reply otherlibs/unix/signals.c]: Well spotted, thanks.  The correct fix uses `caml_unix_error`, see b62ba96d14

File: otherlibs/unix/unixLabels.mli (hunk 3 above)
Comment (xavierleroy) [reply otherlibs/unix/unixLabels.mli]: It was a cut-and-paste error, obviously.  Fixed.

**Review by ghost — APPROVED** [review]
Looking good now.

**Review by MisterDA — APPROVED** [review]

**Comment by xavierleroy** [comment]
Thanks for the reviews and the approvals.  I feel brave enough to merge.
